Make medium slug in maintainer_group model optional

ayu023ban opened the pull request:
### Description
<!--- Describe your changes in detail -->
Fixes #2 
### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

### Screenshots (if appropriate):

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires migration.
- [ ] My change requires migration`.

ayu023ban committed:
Fix: made medium slug in maintainer_group model optional

ayu023ban committed:
resolved 500 error due to empty medium url

ayu023ban commented:
@algomaster99  please review this PR.

algomaster99 commented on models/maintainer_group.py:
```diff
@@ -31,8 +31,8 @@ class MaintainerGroup(Model):
     medium_slug = models.CharField(
         max_length=63,
         blank=True,
+        null = True,
```
No need to add `null=True` because this will be a string. Why?

If you add `null=True`, this field can hold both `NULL` and `''` both of them more or less signify the absence of URL in this context. To avoid this redundancy, let's just stick to `blank=True` so that only `""` signifies empty value.

algomaster99 replied:
There shouldn't be a ` ` (space) before and after `=`.

ayu023ban replied:
done

algomaster99 commented on views/blog.py:
```diff
@@ -36,7 +36,10 @@ def get(self, request, format=None):
+        if response.status_code == 200:
+            required_blog_list = blog_list[:max_blog_count]
+        else:
+            required_blog_list = []
```
Don't send an empty list otherwise. Send an appropriate `Response` like `204: No content`.

ayu023ban replied:
done

algomaster99 commented on models/maintainer_group.py:
```diff
@@ -31,8 +31,8 @@ class MaintainerGroup(Model):
         blank=True,
+        null = True,
     )
-
```
Don't remove this.

ayu023ban replied:
done

algomaster99 reviewed (changes requested):
Nice work :tada: 
Suggested a few changes otherwise looks cool.

Also, have a cursory look over the PEP8 guidelines and try to keep them in mind so that we can keep the code formatting consistent.

ayu023ban committed:
Addressed comments
- Removed null = True in medium slug
- Send 204 status at time of empty medium or invalid medium slug

algomaster99 commented on migrations/0003_auto_20200917_1017.py:
```diff
@@ -0,0 +1,18 @@
+        migrations.AlterField(
+            model_name='maintainergroup',
+            name='medium_slug',
+            field=models.CharField(blank=True, max_length=63, null=True),
```
You forgot to run `python manage.py makemigrations`. Also, make sure you squash the migration generated and this one.

algomaster99 commented on migrations/0004_auto_20200918_2251.py:
```diff
@@ -0,0 +1,18 @@
+        migrations.AlterField(
+            model_name='maintainergroup',
+            name='medium_slug',
+            field=models.CharField(blank=True, default='', max_length=63),
```
No need for `default=''`

ayu023ban replied:
removed

algomaster99 reviewed (changes requested):
The commit messages should start with the present tense. For example: Add condition on rendering medium connecting icon https://github.com/IMGIITRoorkee/omniport-frontend-maintainer-site/pull/28#pullrequestreview-491594457

Please refer to PEP8 guidelines too. The code formatting is not consistent.

ayu023ban committed:
Resolve Message error in http204 status

ayu023ban committed:
- Use AutoPep8 to Lint the code
- Remove default value for medium_slug
- Remove unnecessary migration files

ayu023ban committed:
Revert Permissions of file

ayu023ban commented:
@algomaster99  i have reverted the file permissions please review it again

algomaster99 commented on views/blog.py:
```diff
@@ -1,8 +1,10 @@
 import requests
 import json
 
+from django.http import HttpResponse
```
Why has this been imported?

ayu023ban replied:
i was trying sending message with diffferent types of responses. will remove that

algomaster99 commented on views/blog.py:
```diff
@@ -36,7 +38,10 @@ def get(self, request, format=None):
+        if response.status_code == 200:
+            required_blog_list = blog_list[:max_blog_count]
+        else:
+            return Response(status=status.HTTP_204_NO_CONTENT)
```
Add a message too. "No blog posts exist"

ayu023ban replied:
i tried to add message in the third commit of this pr. but when ever i add some data or object with status 204 then the request doesnot get passed.
i read about it and its only with the case of status 204 as you cannot send any data or(content) with status 204 .
otherwise it wouldn't be ""NO_CONTENT""

algomaster99 replied:
Oh, okay. Makes sense!

algomaster99 reviewed (changes requested):
One last small change

ayu023ban committed:
Remove extra dependency of HTTPResponse

ayu023ban commented:
@algomaster99  please merge this PR

algomaster99 commented:
Will do once [these](https://github.com/IMGIITRoorkee/omniport-frontend-maintainer-site/pull/28#pullrequestreview-496939377) are addressed.
